### ISSUE TITLE
Nullability false positive warnings when assigned to a local variable

### ISSUE THREAD

**Issue by yufeih** [dotnet/roslyn#50250]
**Steps to Reproduce**:

```csharp

static void M(string? x)
{
    if (x != null)
    {
        x.ToString();  // Does not report a warning
    }

    // Doesn't work
    var isNotNull = x != null;
    if (isNotNull)
    {
        x.ToString(); // Warns today, but perhaps shouldn't warn
    }
}
```

[Sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8DEA7AVwwwmAzgAI49TyBYAKEYAEBmC5gJg4EYB2RgG9GFUbwBsHACwUAsgApmPAAwB+CgA8AlCLHCGYwxQBuEBBTRQAcgHsYVohgoBeTRQCErwsQDcuoxYAZhTylrb2jjoGART6MUYaAHQAKjYAyjAIaHgA5vJaPhQA9EUUAO5meFAUMDYAJhAAnkgUwAQwFAAOiAAWEJ3VUD02RHV4AOQdFQh4/kYAvnOLDMtAA==)

**Comment by Youssef1313** [comment]
This was previously stated as by-design. See https://github.com/dotnet/roslyn/issues/38561.
